jfr-mappers/src/main/java/com/newrelic/jfr/toevent/MethodSampleMapper.java

```diff
@@ -55,9 +55,17 @@ public static MethodSampleMapper forNativeMethodSample() {
 
     var timestamp = ev.getStartTime().toEpochMilli();
     var attr = new Attributes();
-    attr.put("thread.name", ev.getThread("sampledThread").getJavaName());
-    attr.put("thread.state", ev.getString("state"));
-    attr.put("stackTrace", MethodSupport.serialize(ev.getStackTrace()));
+    var sampledThread = ev.getThread("sampledThread");
+    if (sampledThread != null && sampledThread.getJavaName() != null) {
+      attr.put("thread.name", sampledThread.getJavaName());
+    }
+    var state = ev.getString("state");
+    if (state != null) {
+      attr.put("thread.state", state);
+    }
+    if (ev.getStackTrace() != null) {
+      attr.put("stackTrace", MethodSupport.serialize(ev.getStackTrace()));
+    }
```

Comment:
Similar to my other suggestion:
```suggestion
    var eventStackTrace = ev.getStackTrace();
    attr.put("stackTrace", eventStackTrace == null ? null : MethodSupport.serialize(eventStackTrace));
```

Reply:
I imagine there is the potential for a stack trace to also exceed the 4K custom event attribute limit. In the telemetry code it might make sense to filter out or truncate extra long values rather than have an almost-silent failure on the server side.